crates/datasources/src/object_store/http.rs

```diff
@@ -23,6 +38,36 @@ pub struct HttpStoreAccess {
     pub url: Url,
 }
 
+impl HttpStoreAccess {
+    async fn content_length(u: Url) -> Result<Option<u64>> {
+        let res = reqwest::Client::new().head(u.clone()).send().await?;
+        let status = res.status();
+        if !status.is_success() {
+            if u.as_str().contains('*') {
+                return Err(ObjectStoreSourceError::InvalidHttpStatus(format!(
+                    "Unexpected status code '{}' for url: '{}'. \
+                    Note that globbing is not supported for HTTP.",
+                    status, u,
+                )));
+            }
```

Comment:
I would have imagined that we'd do this somewhere other than the check for the content length?